base/int.jl

```diff
@@ -332,6 +332,21 @@ for to in BitInteger_types, from in (BitInteger_types..., Bool)
     end
 end
 
+"""
+    rem(x::Integer, T<:Integer)
+    mod(x::Integer, T<:Integer)
+    %(x::Integer, T<:Integer)
+
+Find `y::T` such that `x` ≡ `y` (mod n), where n is the number of integers representable
+in `T`, and `y` is an integer in `[typemin(T),typemax(T)]`.
```

Comment:
Sure, it would make sense semantically. Since nothing is a cheap bit-level operation on BigInts, there's no harm. Since the range of BigInts is unbounded, `n % BigInt` is equivalent to a convert. In the other direction, `b % Int` can be done by doing a single word load.